### PR TITLE
fix: drop usage of pkg_resources

### PR DESCRIPTION
**Pull request by TheKevJames**
Reopening #2419, based on @parthea 's review. Removes setuptools without
introducing importlib-metadata.

Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-aiplatform/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)
- [ ] Get the necessary approvals
- [ ] Once the last commit on the PR has been approved, add the "ready to pull" label to the Pull Request

Note: do not merge your PR from GitHub. Adding the "ready to pull" label is the final step in the review process.
After approvals, the changes in your PR will be committed to the `main` branch and this PR will be closed.

Towards https://github.com/googleapis/python-aiplatform/issues/2931 🦕

### CODE REVIEW

**Comment by TheKevJames** [comment]
@parthea had previously marked:
* @parthea as reviewer
* @srarob and @sasha-gitg as assignee

**Review by parthea — APPROVED** [review]
@TheKevJames , Thanks for the fix!

File: google/cloud/aiplatform/initializer.py [hunk 1]
```diff
@@ -30,6 +29,7 @@
 from google.auth import credentials as auth_credentials
 from google.auth.exceptions import GoogleAuthError
 
+from google.cloud.aiplatform import __version__
```
Comment (nayaknishant) [inline google/cloud/aiplatform/initializer.py:32]: nit: do we want this as 
```
from google.cloud.aiplatform import __version__
gapic_version = __version__
```

or 

```
from google.cloud import aiplatform
gapic_version = aiplatform.__version__
```

**Review by nayaknishant — APPROVED** [review]

**Comment by nayaknishant** [comment]
https://github.com/googleapis/python-aiplatform/commit/32c7197671bda607a978cd2c50598cdd85ecd96b has been published. Closing this PR.

**Pull request closed without merging**